node/forest_libp2p/bitswap/src/request_manager.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::{event_handlers::*, *};
+use ahash::{HashMap, HashMapExt, HashSet, HashSetExt};
+use flume::TryRecvError;
+use libipld::Block;
+use libipld::Cid;
+use libp2p::PeerId;
+use parking_lot::RwLock;
+use std::{
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+const BITSWAP_BLOCK_REQUEST_INTERVAL: Duration = Duration::from_millis(500);
+
+#[derive(Debug, Clone)]
+struct ResponseChannels {
+    block_have: flume::Sender<PeerId>,
+    block_saved: flume::Sender<()>,
+}
+
+// TODO: Use message queue like `go-bitswap`
+#[derive(Debug)]
+pub struct BitswapRequestManager {
+    outbound_request_tx: flume::Sender<(PeerId, BitswapRequest)>,
+    peers: RwLock<HashSet<PeerId>>,
+    response_channels: RwLock<HashMap<Cid, ResponseChannels>>,
+}
+
+impl BitswapRequestManager {
+    pub fn new(outbound_request_tx: flume::Sender<(PeerId, BitswapRequest)>) -> Self {
+        Self {
+            outbound_request_tx,
+            peers: RwLock::new(HashSet::new()),
+            response_channels: RwLock::new(HashMap::new()),
+        }
+    }
+}
+
+impl BitswapRequestManager {
+    pub fn add_peer(&self, peer: PeerId) -> bool {
+        let mut peers = self.peers.write();
+        let r = peers.insert(peer);
```

Comment:
let's use more meaningful names

Reply:
Renamed to `on_peer_connected`

Reply:
I think Hubert is referring to the `r` variable.

Reply:
Oh gocha, fixed